src/log.h

```diff
@@ -249,8 +249,10 @@ typedef enum
     LY_EINT,        /**< Internal error */
     LY_EVALID,      /**< Validation failure */
     LY_EDENIED,     /**< Operation is not allowed */
-    LY_EINCOMPLETE, /**< The operation did not failed, but for some reason it was not possible to finish it completely.
+    LY_EINCOMPLETE, /**< The operation did not fail, but for some reason it was not possible to finish it completely.
                          According to the specific use case, the caller is usually supposed to perform the operation again. */
+    LY_ERECOMPILE,  /**< The operation did not fail, but requires context recompilation before it can be completed.
```

Comment:
so is this value now expected even in API functions adding modules into the context? If so, it should be probably mentioned in the description of `LY_CTX_EXPLICIT_COMPILE`

Reply:
The only API function that can return `LY_ERECOMPILE` should be `lyplg_type_make_implemented()`. In that case it should be treated the same way as any other error value. In all other cases (internally) simply the `recompile` flag in global `unres` is set and it is checked at relevant places.

I wanted to unify the approach but checking some flag (of an internal structure currently) in a type plugin seemed to me like a bad idea, so I have added the error value. So I have tried using the error value through all of *libyang*, which resulted only in me wasting a few hours and it still not working. The problem mostly was that this is not a standard error value and should "bubble up" the stack only up to a certain point, where the context should be recompiled and compilation retried. Also, in the functions that are returning the error, they also needed to often behave specially. And trying to implement all this, there were just more and more functions with messy cleanups and finding new problems was taking too long. So I ended up with the current solution as the best, under the circumstances.

I want to refactor some functions later and ideally use the return value instead of the unres flag but I am not so sure anymore, whether I can come up with anything better.